hare/preroundtracker.go

```diff
@@ -131,9 +131,9 @@ func (pre *preRoundTracker) CanProveSet(set *Set) bool {
 
 // FilterSet filters out non-provable values from the given set.
 func (pre *preRoundTracker) FilterSet(set *Set) {
-	for _, bid := range set.elements() {
-		if !pre.CanProveValue(bid) { // not enough witnesses
-			set.Remove(bid)
+	for _, v := range set.ToSlice() {
+		if !pre.CanProveValue(v) { // not enough witnesses
+			set.Remove(v)
 		}
```

Comment:
Double iteration, with the `ForEachElement` closure this would then be:

```suggestion
	set.ForEachElement(func(v types.ProposalID) {
		if !pre.CanProveValue(v) { // not enough witnesses
			set.Remove(v)
		}
	}
```